isa-bel/react-native-workshop
# Feat - Add components

#### What does this PR do?

Adds the base components for the app. Each component includes tests.

Also adds a commons style file, [plop](https://plopjs.com/) template for new components, and pre-push hooks for linting and testing.

#### Where should the reviewer start?

In `src/components/Icon/Icon.js`, `src/components/TextIcon/TextIcon.js`, and `src/components/ActionsRow/ActionsRow.js` for the new components.

#### How should this be manually tested?
1. Update the dependencies.
    ```bash
    yarn
    ```
2. Run the test script.
    ```bash
    yarn test
    ```
3. Verify that there are no errors.

To test plop:

1. Run plop:
    ```bash
    yarn plop
   ```
2. Name your component.
3. Verify that plop created the new component folder and files.

#### Any background context you want to provide?



#### What are the relevant tickets?



N/A

#### Checklist
